data.php

```diff
@@ -183,10 +185,15 @@ function setShowBlockedPermitted()
     }
 
     function getAllQueries($orderBy) {
-        global $log,$gravity,$showBlocked,$showPermitted;
+        global $log,$gravity,$showBlocked,$showPermitted,$whitelist,$blacklist;
         $allQueries = array("data" => array());
         $dns_queries = getDnsQueriesAll($log);
-        $gravity_domains = getGravityDomains($gravity);
+        $gravity_domains = getDomains($gravity, true);
+        $whitelist_domains = getDomains($whitelist, false);
+        $blacklist_domains = getDomains($blacklist, true);
+
+        $gravity_domains = array_merge($gravity_domains, $whitelist_domains, $blacklist_domains);
```

Comment:
There is really no way to avoid having three arrays, since we have to read in the gravity domains, whitelist, and blacklist.